### PR TITLE
Not saving null post categories

### PR DESCRIPTION
**Pull request by tonyr59h**
Addresses #3048.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java [hunk 1]
```diff
@@ -472,9 +472,12 @@ else if (mPost.getDate_created_gmt() > 0)
             mPost.setLocation(mPostLocation);
         }
 
+        if (mCategories != null) {
+            mPost.setJSONCategories(new JSONArray(mCategories));
+        }
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java]: Should the post categories be cleared/nulled when `mCategories` is null?

Also, there are a number of other places in this fragment where `mCategories` is accessed without a null check ([for example](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java#L307)). Should these be addressed as well?

Comment (tonyr59h) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java]: I'm trying to figure out how it got to be null in the first place as it's never explicitly set to null. If it's just in a bad state I don't think we'd want to delete the existing categories. All I've got to go off is that `updatePostSettings` was called from the auto-save background timer in `EditPostActivity`.

Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostSettingsFragment.java]: > If it's just in a bad state I don't think we'd want to delete the existing categories.

Fair enough. But what about adding null checks in the other places where `mCategories` is accessed?

**Comment by tonyr59h** [comment]
Checking for null everywhere else now. One of the null checks occurs on a button click so I also added a toast to let the user know why nothing happens. Wording on the toast sounds a bit too technical, I'm open to suggestions :)

**Comment by nbradbury** [comment]
@tonyr59h The changes look good. I almost hate to say this, but I think the "corrupt categories" error will be so rare that it's fine to just use `R.string.error_generic` for the toast rather than have a separate string for it.

**Comment by tonyr59h** [comment]
> I almost hate to say this, but I think the "corrupt categories" error will be so rare that it's fine to just use R.string.error_generic for the toast rather than have a separate string for it.

This is actually good to hear :relieved: I didn't really like the error string I put in, I'm totally on board with the generic string.

**Comment by nbradbury** [comment]
Looks good :shipit:

**Pull request merged**